spec/requests/api/firmwares_spec.rb

```diff
@@ -0,0 +1,31 @@
+RSpec.describe "firmwares API" do
+  describe "display firmware details" do
+    context "with a valid role" do
+      it "shows its properties" do
+        fw = FactoryGirl.create(:firmware, :id => 1, :name => "UEFI",
```

Comment:
You probably don't want to set the `id` here - that's the database's job and could result in some erratic behavior